hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpCommandParser.java

```diff
@@ -37,7 +39,16 @@ public final TextCommand parser(TextDecoder decoder, String cmd, int space) {
         } else {
             return new ErrorCommand(ERROR_CLIENT);
         }
-        return createHttpCommand(decoder, uri);
+        try {
+            String urlDecodedUri = decodeUrl(uri);
+            return createHttpCommand(decoder, urlDecodedUri);
+        } catch (UnsupportedEncodingException e) {
+            return new ErrorCommand(ERROR_CLIENT, "URL is not encoded properly");
+        }
+    }
+
+    private static String decodeUrl(String value) throws UnsupportedEncodingException {
```

Comment:
minor: do we need a separate function for this one-liner, that is not reused?

Reply:
removed